server/src/main/java/io/crate/expression/scalar/ToCharFunction.java

```diff
@@ -0,0 +1,162 @@
+
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Locale;
+import java.util.TimeZone;
+import java.util.function.BiFunction;
+
+public class ToCharFunction extends Scalar<String, Object> {
+
+    public static final String NAME = "to_char";
+
+    public static void register(ScalarFunctionModule module) {
+        List.of(DataTypes.TIMESTAMP, DataTypes.TIMESTAMPZ).stream()
+            .forEach(type -> {
+                module.register(
+                    Signature.scalar(
+                        NAME,
+                        type.getTypeSignature(),
+                        DataTypes.STRING.getTypeSignature(),
+                        DataTypes.STRING.getTypeSignature()
+                    ),
+                    (signature, boundSignature) ->
+                        new ToCharFunction(
+                            signature,
+                            boundSignature,
+                            type,
+                            ToCharFunction::evaluateTimestamp
+                        )
+                );
+            });
+
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.INTERVAL.getTypeSignature(),
+                DataTypes.STRING.getTypeSignature(),
+                DataTypes.STRING.getTypeSignature()
+            ),
+            (signature, boundSignature) ->
+                new ToCharFunction(
+                    signature,
+                    boundSignature,
+                    DataTypes.INTERVAL,
+                    ToCharFunction::evaluateInterval
+                )
+        );
+
+        DataTypes.NUMERIC_PRIMITIVE_TYPES.stream()
+            .forEach(type -> {
+                module.register(
+                    Signature.scalar(
+                        NAME,
+                        type.getTypeSignature(),
+                        DataTypes.STRING.getTypeSignature(),
+                        DataTypes.STRING.getTypeSignature()
+                    ),
+                    (signature, boundSignature) ->
+                        new ToCharFunction(
+                            signature,
+                            boundSignature,
+                            type,
+                            ToCharFunction::evaluateNumber
+                        )
+                );
+            });
+    }
+
+    private final Signature signature;
+    private final Signature boundSignature;
+    private final DataType expressionType;
+    private final BiFunction<Object, String, String> evaluatorFunc;
+
+    public ToCharFunction(Signature signature, Signature boundSignature, DataType expressionType, BiFunction<Object, String, String> evaluatorFunc) {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+        this.expressionType = expressionType;
```

Comment:
you do not really have to pass the argument type explicitly, you can already get it from the bound signature.
```
assert boundSignature.getArgumentDataTypes().size() == 1
boundSignature.getArgumentDataTypes().get(0)
```